### ISSUE TITLE
When failing due to a templating error, the error message is unclear and ansible does not exit with a non-zero status code

### ISSUE THREAD

**Issue by crisvp** [fboender/ansible-cmdb#94]
When it encounters a KeyError while processing it reports the failure as "address".

```
> ansible-cmdb -d -t keyerror.tpl.txt /tmp/fnord
data_dir = /usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data
tpl_dir = /usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data/tpl
static_dir = /usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data/static
config_file = None
inventory hostfile = None
Parsing fact dir: /tmp/fnord
Reading host facts from /tmp/fnord/{bunch of hosts}
template file = keyerror.tpl.txt
Template params: {'lib_dir': '/usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data', 'version': '1.18', 'data_dir': '/usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data', 'log': <logging.RootLogger object at 0x106747810>, 'columns': None}


Traceback (most recent call last):
  File "/usr/local/bin/ansible-cmdb", line 147, in <module>
    out = mytemplate.render(hosts=ansible.hosts, **params)
  File "/usr/local/Cellar/ansible-cmdb/1.18/mako/template.py", line 443, in render
    return runtime._render(self, self.callable_, args, data)
  File "/usr/local/Cellar/ansible-cmdb/1.18/mako/runtime.py", line 803, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/local/Cellar/ansible-cmdb/1.18/mako/runtime.py", line 835, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/local/Cellar/ansible-cmdb/1.18/mako/runtime.py", line 860, in _exec_template
    callable_(context, *args, **kwargs)
  File "keyerror.tpl.txt", line 173, in render_body
    <td>${col["func"](host)}</td>
  File "keyerror.tpl.txt", line 0, in col_main_ip

  File "keyerror.tpl.txt", line 53, in render_col_main_ip
    ${host['ansible_facts']['ansible_default_ipv4']['address']}
KeyError: 'address'

Whoops, it looks like something went wrong while rendering the template.

The reported error was: 'address'

The output is probably not correct.
You can report a bug on the issue tracker:

  https://github.com/fboender/ansible-cmdb/issues

Please include the debugging output (-d switch) in the report!

If you can, also include the hosts file and the facts file for the last host
that rendered properly ('Rendering host...' in the output. If these files must
remain confidential, you can send them to ferry.boender@gmail.com instead.
> echo $?
0
```

The process then exits with a zero exit-status, instead of the expected non-zero status.

Under different circumstances the message appears missing entirely:

```
> ansible-cmdb -d -t ./blankerror.tpl -i production/inventory /tmp/fnord
\data_dir = /usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data
tpl_dir = /usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data/tpl
static_dir = /usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data/static
config_file = None
inventory hostfile = production/inventory
Parsing fact dir: /tmp/fnord
Reading host facts from {bunch of hosts}
Parsing host vars (dir): production/inventory
template file = ./blankerror.tpl
Template params: {'lib_dir': '/usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data', 'version': '1.18', 'data_dir': '/usr/local/Cellar/ansible-cmdb/1.18/ansiblecmdb/data', 'log': <logging.RootLogger object at 0x10287c890>, 'columns': None}
















> echo $?
0
```

Again, with zero exit status.

Tested on ansible-cmdb 1.18 on MacOS.
The (obviously broken) template files are attached.

[blankerror.tpl.txt](https://github.com/fboender/ansible-cmdb/files/660884/blankerror.tpl.txt)
[keyerror.tpl.txt](https://github.com/fboender/ansible-cmdb/files/660885/keyerror.tpl.txt)

**Comment by fboender** [comment]
> When it encounters a KeyError while processing it reports the failure as "address".

I've made two changes in master. One shows the full stack trace even without specifying "-d". The other one also shows the name of the thrown exception (KeyError in this case). This should make it slightly clearer. Of course the best solution would be to show the actual host that is missing the specific value, but this is something that can't be detected outside of the template. The best way to go about it is to use the "jsonxs" library, which doesn't care about missing keys / attributes. Check the lastest "html_fancy" template in master for details on how to use it.

> The process then exits with a zero exit-status, instead of the expected non-zero status.

I've made a commit to fix this. If an error occurs during rendering, ansible-cmdb now exits with status code 1.

> Under different circumstances the message appears missing entirely:

This appears to be because nothing actually went wrong. Your template 'blankerror.tpl.txt' only defines functions, but never actually calls them. Hence no output, no error and an exit status of 0.
